website/client/src/store/getters/tasks.js

```diff
@@ -34,12 +34,66 @@ function getTaskColor (task) {
   return 'best';
 }
 
-export function canDelete () {
-  return task => {
-    const isUserChallenge = Boolean(task.userId);
-    const activeChallenge = isUserChallenge
-      && task.challenge && task.challenge.id && !task.challenge.broken;
-    return !activeChallenge;
+export function canDelete (store) {
+  return (task, taskCategory, isTaskFromUserDashboard, group, challenge) => {
+    let isUserCanDeleteTask = isTaskFromUserDashboard;
+    const user = store.state.user.data;
+
+    switch (taskCategory) {
+      case 'challenge':
+        if (challenge && !isTaskFromUserDashboard) {
+          const isUserChallenge = user.id === challenge.leader.id;
+          isUserCanDeleteTask = isUserChallenge || task.challenge.broken;
```

Comment:
if we're not on the user task board, the challenge is broken but you're not the leader I don't think that the `delete` action should be show as it would be on the challenge's task board anyway. Correct, @Alys ?

Reply:
Correct, a challenge participant should never see the delete button on the challenge's task board.

@Xaz16 I don't know if this is relevant since the issue is just about the user's own task board, but if you are changing code that affects the challenge's task board, be aware that moderators/staff have the same edit/delete rights as challenge owners (i.e., a moderator should see a delete button on the challenge's task board, just as the challenge leader does). Here's an example of how to identify moderators along with challenge leaders:
https://github.com/HabitRPG/habitica/blob/1859f431338e16ca9fa2a4245bfe8dd068207315/website/server/models/challenge.js#L80-L83